packages/mwp-api-state/src/sync/index.js

```diff
@@ -74,19 +74,29 @@ export const getNavEpic = (routes, baseUrl) => {
 				// note that this function executes _downstream_ of reducers, so the
 				// new `routing` data has already been populated in `state`
 				const state = store.getState();
+				const { referrer = {} } = state.routing;
 				// inject request metadata from context, including `store.getState()`
 				const requestMetadata = {
-					referrer: (state.routing.referrer || {}).pathname || '',
+					referrer: referrer.pathname || '',
 					logout: location.pathname.endsWith('logout'), // assume logout route ends with logout
 					clickTracking: state.clickTracking,
+					retainRefs: [],
 				};
 
 				const cacheAction$ = requestMetadata.logout
 					? Observable.of({ type: 'CACHE_CLEAR' })
 					: Observable.empty();
 
 				const apiAction$ = Observable.fromPromise(
-					resolveRoutes(location, baseUrl).then(getMatchedQueries(location))
+					Promise.all([
```

Comment:
So if i'm reading this right, if i navigate 
group home w/ `retainref` for group query <- makes query request
navigate to child route group discussion page <- no new request
go back group home <- makes query request

right? :)

Reply:
code-wise all of this should be completely invisible to consumer apps - it will work automagically without you needing to specify anything in your queries

specifically, this routine programmatically checks what refs were populated by the queries on the previous page, diffs them with the refs that will be populated by the current page, and _only_ clears the _new_ queries.

It does _not_ prevent a new request from being made to the API - all refs will still be updated, but the 'stable' ones won't be cleared out the way they are now. When the API returns, the 'stable' values will be updated, but it is expected that there will be minimal changes - the app will re-render, but visually not much should change

Reply:
But yes, dynamic refs should definitely prevent stale event data from ever showing up :)

Reply:
> It does not prevent a new request from being made to the API - all refs will still be updated, but the 'stable' ones won't be cleared out the way they are now. When the API returns, the 'stable' values will be updated, but it is expected that there will be minimal changes - the app will re-render, but visually not much should change

![giphy](https://user-images.githubusercontent.com/1933722/31523189-0925ceb2-af80-11e7-823b-1f6529295ca7.gif)